src/legacy/status_im/log_level/core.cljs

```diff
@@ -1,51 +1,35 @@
 (ns legacy.status-im.log-level.core
   (:require
     [legacy.status-im.multiaccounts.update.core :as multiaccounts.update]
-    [re-frame.core :as re-frame]
-    [taoensso.timbre :as log]
-    [utils.i18n :as i18n]
+    [native-module.core :as native-module]
     [utils.re-frame :as rf]))
 
+(rf/reg-fx
+ :log-level/set-log-level
+ (fn [log-level]
+   (when (seq log-level)
+     (native-module/set-log-level log-level))))
+
+(rf/reg-fx
+ :log-level/set-log-enabled
+ (fn [enabled?]
+   (native-module/set-log-enabled enabled?)))
+
 (rf/defn save-log-level
   {:events [:log-level.ui/change-log-level-confirmed]}
-  [{:keys [db]} log-level]
+  [{:keys [db] :as cofx} log-level]
   (let [old-log-level (get-in db [:profile/profile :log-level])]
-    (when (not= old-log-level log-level)
+    (if (not= old-log-level log-level)
       (let [need-set-log-enabled? (or (empty? old-log-level) (empty? log-level))
-            log-enabled?          (boolean (seq log-level))
-            rpc-calls             (cond-> []
-                                    log-enabled?
-                                    (conj {:method   "wakuext_setLogLevel"
-                                           :params   [{:logLevel log-level}]
-                                           :on-error #(log/error "Failed to set log level" %)})
-
-                                    need-set-log-enabled?
-                                    (conj {:method   "wakuext_setLogEnabled"
-                                           :params   [log-enabled?]
-                                           :on-error #(log/error "Failed to set log enabled" %)}))]
-        {:fx [[:json-rpc/call
-               (conj (vec (map #(assoc % :on-success nil) (butlast rpc-calls)))
-                     (assoc (last rpc-calls)
-                            :on-success
-                            #(rf/dispatch [:log-level/update-multiaccount log-level])))]]}))))
-
-(rf/defn update-multiaccount
-  {:events [:log-level/update-multiaccount]}
-  [cofx log-level]
-  (multiaccounts.update/multiaccount-update
-   cofx
-   :log-level
-   log-level
-   {:on-success #(rf/dispatch [:profile/logout])}))
-
-(rf/defn show-change-log-level-confirmation
-  {:events [:log-level.ui/log-level-selected]}
-  [_ {:keys [name value]}]
-  {:ui/show-confirmation
-   {:title               (i18n/label :t/close-app-title)
-    :content             (i18n/label :t/change-log-level
-                                     {:log-level name})
-    :confirm-button-text (i18n/label :t/close-app-button)
-    :on-accept           #(re-frame/dispatch
-                           [:log-level.ui/change-log-level-confirmed value])
-    :on-cancel           nil}})
+            log-enabled?          (boolean (seq log-level))]
+        (merge
```

Comment:
@qfrank Here we can simplify, similarly to how I shared in the code diff in https://github.com/status-im/status-mobile/pull/22275#pullrequestreview-2678471107

Basically if you add unique a re-frame event ID to `multiaccount-update` you can dispatch to it and then you can get rid of the mixed approach with `merge`. In the diff you can also see the third argument is optional, so you should be able to remove `{:on-success #()}`.